test/libsyncengine/update_detection/file_system_observer/testsnapshot.cpp

```diff
@@ -45,101 +39,214 @@ void TestSnapshot::tearDown() {}
+    CPPUNIT_ASSERT(_snapshot->parentId("aa") == "b");
+    _snapshot->getChildrenIds("b", childrenIds);
     CPPUNIT_ASSERT(childrenIds.contains("aa"));
-    snapshot.getChildrenIds("a", childrenIds);
+    _snapshot->getChildrenIds("a", childrenIds);
     CPPUNIT_ASSERT(childrenIds.empty());
 
     // Remove node B
-    snapshot.removeItem("b");
-    snapshot.getChildrenIds(rootNodeId, childrenIds);
-    CPPUNIT_ASSERT(!snapshot.exists("aaa"));
-    CPPUNIT_ASSERT(!snapshot.exists("aa"));
-    CPPUNIT_ASSERT(!snapshot.exists("b"));
+    _snapshot->removeItem("b");
+    _snapshot->getChildrenIds(_rootNodeId, childrenIds);
+    CPPUNIT_ASSERT(!_snapshot->exists("aaa"));
+    CPPUNIT_ASSERT(!_snapshot->exists("aa"));
+    CPPUNIT_ASSERT(!_snapshot->exists("b"));
     CPPUNIT_ASSERT(!childrenIds.contains("b"));
 
     // Reset snapshot
-    snapshot.init();
-    CPPUNIT_ASSERT_EQUAL(static_cast<uint64_t>(1), snapshot.nbItems());
+    _snapshot->init();
+    CPPUNIT_ASSERT_EQUAL(static_cast<uint64_t>(1), _snapshot->nbItems());
 }
 
 void TestSnapshot::testSnapshotInsertionWithDifferentEncodings() {
-    const NodeId rootNodeId = *SyncDb::driveRootNode().nodeIdLocal();
-
-    const DbNode dummyRootNode(0, std::nullopt, Str("Local Drive"), SyncName(), "1", "1", std::nullopt, std::nullopt,
-                               std::nullopt, NodeType::Directory, 0, std::nullopt);
-    Snapshot snapshot(ReplicaSide::Local, dummyRootNode);
-
-    const SnapshotItem nfcItem("A", rootNodeId, testhelpers::makeNfcSyncName(), 1640995201, -1640995201, NodeType::Directory, 123,
-                               false, true, true);
-    const SnapshotItem nfdItem("B", rootNodeId, testhelpers::makeNfdSyncName(), 1640995201, -1640995201, NodeType::Directory, 123,
-                               false, true, true);
+    const SnapshotItem nfcItem("A", _rootNodeId, testhelpers::makeNfcSyncName(), testhelpers::defaultTime,
+                               -testhelpers::defaultTime, NodeType::Directory, testhelpers::defaultFileSize, false, true, true);
+    const SnapshotItem nfdItem("B", _rootNodeId, testhelpers::makeNfdSyncName(), testhelpers::defaultTime,
+                               -testhelpers::defaultTime, NodeType::Directory, testhelpers::defaultFileSize, false, true, true);
     {
-        snapshot.updateItem(nfcItem);
+        _snapshot->updateItem(nfcItem);
         SyncPath syncPath;
-        snapshot.path("A", syncPath);
+        bool ignore = false;
+        _snapshot->path("A", syncPath, ignore);
         CPPUNIT_ASSERT_EQUAL(SyncPath(testhelpers::makeNfcSyncName()), syncPath);
     }
     {
-        snapshot.updateItem(nfdItem);
+        _snapshot->updateItem(nfdItem);
         SyncPath syncPath;
-        snapshot.path("B", syncPath);
+        bool ignore = false;
+        _snapshot->path("B", syncPath, ignore);
         CPPUNIT_ASSERT_EQUAL(SyncPath(testhelpers::makeNfdSyncName()), syncPath);
     }
 }
 
+void TestSnapshot::testPath() {
+    {
```

Comment:
It would be nice to have a short comment explaining the purpose and the expected outcome of each subtest.

Reply:
Done in 6c66776ee849ef7120694f5f3064e21c10907145